frontend/pages/events.js

```diff
@@ -1,20 +1,196 @@
+    const token = tokenCtx || localStorage.getItem("token");
+    const AuthStr = `Bearer ${token}`;
+    const today = new Date().toISOString();
+
+    // Default fetch is any event from today with a limit of 10
+    const eventsPromise = axios({
+      method: "get",
+      url: `http://localhost:8000/events?compare=gt&date_from=${today}&limit=10`,
+      headers: {
+        Authorization: AuthStr
+      }
+    });
+
+    const placesPromise = axios({
+      method: "get",
+      url: `http://localhost:8000/places`,
+      headers: {
+        Authorization: AuthStr
+      }
+    });
+
+    const activitiesPromise = axios({
+      method: "get",
+      url: `http://localhost:8000/activities`,
+      headers: {
+        Authorization: AuthStr
+      }
+    });
+    const [events, places, activities] = await Promise.all([eventsPromise, placesPromise, activitiesPromise]);
+
+    this.setState({ events: events.data.events, places: places.data.places, activities: activities.data.activities });
+  }
+
+  updateFilter = (type, data) => {
+    const oldState = Object.assign({}, this.state);
+    const oldFilters = oldState.eventFilters;
+    oldFilters[type] = data;
+    this.setState({ eventFilters: oldFilters });
+  };
+
+  clearFilters = () => {
+    this.setState({ eventFilters: { datefrom: null, city: null, activity: null } });
+  };
+
   render() {
+    const { events, places, activities, eventFilters } = this.state;
     return (
-      <Container>
-        <h1>Dashboard</h1>
-        <EventList />
-      </Container>
+      <MainLayout>
+        <TopPanel>
+          <div>
+            <h4>Browse existing events</h4>
+            <BrowseImg src=".././static/browsing.png" alt="globe-search" />
+          </div>
+          <div>
+            <h4>or</h4>
+            <ChoiceImg src=".././static/choice1.png" alt="person-choice" />
+          </div>
+          <div>
+            <h4>
+              <Link href="/newevent">
+                <StyledButton>Create</StyledButton>
+              </Link>{" "}
+              your own and PairUp!
+            </h4>
+            <CreateImg src=".././static/people.png" alt="people" />
+          </div>
+        </TopPanel>
+        <Divider>
+          <h4>
+            <span>Active Events</span>
+          </h4>
+        </Divider>
+        <FilterControlPanel>
+          <span>Pick a</span>
+          <ActivityPicker activities={activities} updateFilter={this.updateFilter} />
+          <span>choose</span>
+          <DateSelectorDynamic updateFilter={this.updateFilter} />
+          <span>and</span>
+          <LocationSearch locations={places} updateFilter={this.updateFilter} />
+          <button type="button" onClick={this.clearFilters}>
+            Clear
+          </button>
+        </FilterControlPanel>
+        {events && <EventList events={events} filters={eventFilters} />}
+      </MainLayout>
     );
   }
 }
 
+Dashboard.contextType = UserContext;
```

Comment:
Interesting, I didn't see this class property used before. Does it mean we don't need to have the ContextConsumer tag in the parent component of the page?

Reply:
Yes I believe we would not need the consumer. This way context can be access across all lifecycle methods with `this.context` which is very handy. Never used it before but seems like a quality of life type of improvement.